### PR TITLE
Feature: PHP8.3 and Kirby3.10 compatible

### PR DESCRIPTION
**Pull request by ceesvanegmond**
Fixes #

#337

## Proposed Changes

- Feature: Be compatible with latest Kirby 3.10 (< 4.0)
- Feature: Be compatible with the whole PHP8 series (also 8.3)

Already tagged the package with 4.5.0; tested it with several projects and works fine!

### CODE REVIEW

**Comment by distantnative** [comment]
Can't merge this into `main` - could you try pointing it on the new `kirby-3` branch instead?

**Comment by ceesvanegmond** [comment]
@distantnative Yes!

File: index.php [hunk 1]
```diff
@@ -18,7 +18,7 @@
 // validate Kirby version
 if (
     version_compare(Kirby::version() ?? '0.0.0', '3.7.4', '<') === true ||
-    version_compare(Kirby::version() ?? '0.0.0', '3.10.0', '>=') === true
+    version_compare(Kirby::version() ?? '0.0.0', '4.0', '>=') === true
 ) {
     throw new Exception('Retour 4.4.0 supports Kirby 3.7.4 to 3.9.x');
```
Comment (distantnative) [inline index.php:23]:
```suggestion
    throw new Exception('Retour 4.5.0 supports Kirby 3.7.4 to 3.x');
```

File: composer.json [hunk 2]
```diff
@@ -3,17 +3,17 @@
...
+    "version": "4.5.0",
     "authors": [
         {
             "name": "Nico Hoffmann",
             "email": "on-github@nhoffmann.com"
         }
     ],
     "require": {
-        "php": ">=8.0.0 <8.3.0",
+        "php": "^8.0",
```
Comment (distantnative) [inline composer.json:14]: I don't think this is a good idea - we don't know what breaking changes PHP 8.4 could bring and if this version works with it. So if PHP 8.3 support should be allowed (and you tested it that it works), I'd change this to `<8.4.0`.

**Review by distantnative — CHANGES_REQUESTED** [review]
Two comments that I think we need to fix first.

**Comment by ceesvanegmond** [comment]
@distantnative 👍 Done

**Pull request merged**